### PR TITLE
cmd,pkg: Deflate unnecessary conditionals

### PR DESCRIPTION
**Pull request by timflannagan**
Remove unneeded if/else conditionals where the if/else both specify
return statements.

Deflate if/else conditionals where in loops using the `continue`
keyword.

Switch the if/else logic to avoid nested conditionals.

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/lib/bundle/utils_test.go [hunk 1]
```diff
@@ -70,8 +70,7 @@ func clearDir(dir string) {
 	for _, item := range items {
 		if item.IsDir() {
 			continue
-		} else {
-			os.Remove(filepath.Join(dir, item.Name()))
 		}
+		os.Remove(filepath.Join(dir, item.Name()))
```
Comment (timflannagan) [inline pkg/lib/bundle/utils_test.go:74]: There's already a continue in that first conditional check, so we don't need to wrap file removal logic and can remove that branch entirely.

**Review by joelanford — APPROVED** [review]
/lgtm

File: pkg/lib/registry/registry.go [hunk 2]
```diff
@@ -150,46 +150,45 @@ func populate(ctx context.Context, loader registry.Load, graphLoader registry.Gr
...
-					unpack := <-unpacked
-					if unpack.err != nil {
-						return unpack.err
-					}
-					overwriteImageMap[img.Bundle.Package][unpack.to] = unpack.from
-					if _, ok := unpackedImageMap[unpack.to]; ok {
-						delete(unpackedImageMap, unpack.to)
+			if overwritten == "" {
+				return fmt.Errorf("index add --overwrite-latest is only supported when using bundle images")
+			}
```
Comment (timflannagan) [inline pkg/lib/registry/registry.go:155]: Just swapped the if/else conditional checks so we can avoid an unnecessary nested block of code.

Previously, this block looked like this:

```go
if overwritten != "" {
...
} else {
    return fmt.Errorf("index add --overwrite-latest is only supported when using bundle images")
}
```

We can instead just handle that case where the `overwritten` variable is an empty string and exit early, which allows us to deflate the logic for handling the non-empty scenario and improve readability.

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Comment by timflannagan** [comment]
It looks like codecov isn't overrideable, which is unfortunate. We already have the required labels here and I don't want to have the bot continue to retest while I try and properly update the codecov. Going to force this in.

**Pull request merged**